docs/reference/spec/host-capabilities/06-kubernetes.md

```diff
@@ -122,3 +135,49 @@ Use the `list_resources_all` when dealing with cluster-wide resources.
 #### Output
 
 Result of `GET /apis/$api_version/namespaces/$namespace/$kind/$name`
+
+### Operation - `can_i`
+
+This operation determines if a specific user has the permission to perform an
+action on a Kubernetes resource. It does this by making a
+[`SubjectAccessReview`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.32/#subjectaccessreview-v1-authorization-k8s-io)
+request to the Kubernetes API server.
+
+#### Input
+
+The `can_i` operation requires a JSON object with the following parameters:
+
+```hcl
+{
+  # A JSON-serialized SubjectAccessReview object that defines the user, action, and resource to be checked.
+  "subject_access_review": {
+    "apiVersion": "authorization.k8s.io/v1",
+    "kind": "SubjectAccessReview",
+    "spec": {
+      "user": "system:serviceaccount:mynamespace:myuser",
+      "resourceAttributes": {
+        "namespace": "",
+        "verb": "list",
+        "resource": "secrets"
+      }
+    }
+  } ,
```

Comment:
I would not embed the actual Kubernetes object into our request:

- It ties us to a specific version of a Kubernetes resource
- We send extra useless data between the host and the guest (it's a silly optimization thing, but still)
- We are not consistent with the naming of the variables. This is ugly and error prone. I know this is something the end user is not seeing, but someone writing a SDK could run into issues because some things are camelCase (`apiVersion`) while others are snake case (`subject_access_review`).

I would create our own payload that is snake case and contains only the data we need.

Reply:
Opened an PR to remove the wrapper object: https://github.com/kubewarden/policy-evaluator/pull/721